driver/csiplugin/controllerserver.go

```diff
@@ -275,80 +293,129 @@ func (cs *ScaleControllerServer) createFilesetBasedVol(scVol *scaleVolume) (stri
 		}
 		opt[connectors.UserSpecifiedInodeLimit] = strconv.FormatUint(inodeLimit, 10)
 	}
-	if scVol.ParentFileset != "" {
-		opt[connectors.UserSpecifiedParentFset] = scVol.ParentFileset
+
+	if isNewVolumeType {
+		// For new storageClass first create independent fileset if not present
+		indepFilesetName := scVol.ConsistencyGroup
+		glog.V(4).Infof("creating independent fileset for new storageClass with fileset name: [%v]", indepFilesetName)
+		opt[connectors.UserSpecifiedFilesetType] = independentFileset
+		opt[connectors.UserSpecifiedParentFset] = ""
+		scVol.ParentFileset = ""
+		createDataDir := false
+		filesetPath, err := cs.createFilesetVol(scVol, indepFilesetName, fsDetails, opt, createDataDir)
```

Comment:
To create an independent fileset, instead of `createFilesetVol` here, should we use ` sc.CreateFileset` similar to the one in `CreatePrimaryFileset` function? as the independent fileset is not going to be a volume.

Reply:
createFilesetVol does have multiple checks and takes care of linking fileset etc after fileset creation with sc.CreateFileset. Hence used it. With using sc.CreateFileset() directlt we won't get those advantages.